axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/bankorder/BankOrderLineOriginServiceImpl.java

```diff
@@ -62,7 +62,13 @@ protected String computeRelatedToSelectName(Model model) {
       return ((PaymentScheduleLine) model).getName();
 
     } else if (model instanceof InvoiceTerm) {
-      return ((InvoiceTerm) model).getName();
+      InvoiceTerm invoiceTerm = ((InvoiceTerm) model);
+      if (invoiceTerm.getMoveLine() != null && invoiceTerm.getMoveLine().getMove() != null) {
+        return invoiceTerm.getMoveLine().getOrigin() != null
+            ? invoiceTerm.getMoveLine().getOrigin()
+            : invoiceTerm.getMoveLine().getMove().getReference();
+      }
+      return invoiceTerm.getName();
     } else if (model instanceof Reimbursement) {
```

Comment:
I think a bankOrder can't be related to an Invoice anymore because now we use the InvoiceTerm, so please remove the related code and selection option